src/shared/platform/lind_platform.c

```diff
@@ -240,89 +249,94 @@ int ParseResponse(PyObject* response, int* isError, int* code, char** dataOrMess
+        response = CallPythonFunc(py_context, "LindSyscall", callArgs); \
+        ParseResponse(response, &_isError, &_code, &_data, &_len);      \
+        errno = _isError ? _code : 0;                                   \
+        retval = _isError ? -1 : _code;                                 \
+        UNREFERENCED_PARAMETER(_offset)
+
+#define LIND_API_PART3                                                  \
+        goto cleanup;                                                   \
+        error:                                                          \
+            PyErr_Print();                                              \
+        cleanup:                                                        \
+            Py_XDECREF(callArgs);                                       \
+            Py_XDECREF(response);                                       \
+            PyGILState_Release(gstate);                                 \
+            return retval
+
+#define COPY_DATA(var, maxlen)                                          \
+        if (!_isError) {                                                \
+            assert(_len<=(int)(maxlen));                                \
+            if(var) {                                                   \
+                assert(_data!=NULL);                                    \
+                memcpy((var), _data, _len);                             \
+            }                                                           \
         }
 
-#define COPY_DATA_OFFSET(var, maxlen, total, current) \
-        if(!_isError) { \
-            assert(((int*)_data)[(current)]<=(int)(maxlen));\
-            if(var) { \
-                assert(_data!=NULL); \
-                memcpy((var), _data+sizeof(int)*(total)+_offset, ((int*)_data)[(current)]); \
-            } \
-        } \
-        _offset += ((int*)_data)[(current)];
+#define COPY_DATA_OFFSET(var, maxlen, total, current)                   \
+        if (_data) {                                                    \
+            if (!_isError) {                                            \
+                assert(((int*)_data)[(current)]<=(int)(maxlen));        \
+                if (var) {                                              \
+                    assert(_data!=NULL);                                \
+                    memcpy((var),                                       \
+                           _data + sizeof(int) * (total) + _offset,     \
+                           ((int *)_data)[(current)]);                  \
+                }                                                       \
+            }                                                           \
+            _offset += ((int*)_data)[(current)];                        \
+        }
 
-#define DUMP_DATA(x) printf(#x" = 0x%"NACL_PRIX64"\n", (uint64_t)(x));
+#define DUMP_DATA(x) printf(#x" = %#lx\n", (uint64_t)(x))
```

Comment:
Is there a reason for changing the NaCl default data format?

Reply:
After some examination, it turns this macro isn't even useful, as it's only used in an `#if 0` section.  Removing the dead code.